simple_network_sim/network_of_populations/__init__.py

```diff
@@ -584,6 +583,8 @@ def _computeInfectiousContactsStochastic(
     numberOfContacts = random_state.poisson(contacts, size=int(infectious))
     # If 0, the hypergeometric calls fails, but mathematically is 0
     numberOfContacts = numberOfContacts[numberOfContacts > 0]
+    # Number of contacts cannot be higher than the number of people in the node
+    numberOfContacts = np.minimum(numberOfContacts, int(totalInAge))
```

Comment:
Actually, why can't the number of contacts be bigger than the population? One person can contact more than one other people. I think the actual limit is `totalInAge**2`, isn't it?

Reply:
I think as per the citation you found in a previous PR, the contacts in the mixing matrix are "unique people in a certain age group that you have been in contact with". So each infected person had on average n contacts with people in age group. So if there are 100 people in some age group, and you are a person, the max number of contacts you can report is 100-1.

So technically I should cap it at `int(totalInAge)-1`

Reply:
Your comment would be correct if we summed up all the contacts before passing in the hypergeom. But the `numberOfContacts` is an array of contacts per each person. Hence the `n-1` cap. 

Does that make sense?

Reply:
Ah, yeah, that makes sense.